### ISSUE TITLE
[iqiyi] Unable to log in, with KeyError

### ISSUE THREAD

**Issue by ckckchen** [rg3/youtube-dl#9276]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like that [x])
- Use *Preview* tab to see how your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2016.04.19*. If it's not read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x ] I've **verified** and **I assure** that I'm running youtube-dl **2016.04.19**

### Before submitting an *issue* make sure you have:
- [x ] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [ x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [x ] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add `-v` flag to **your command line** you run youtube-dl with, copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):
```
youtube-dl --verbose -u 18600775053 -p woaiwojia521 http://www.iqiyi.com/v_19rrlrh3yg.html
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'--verbose', u'-u', u'PRIVATE', u'-p', u'PRIVATE', u'http://www.iqiyi.com/v_19rrlrh3yg.html']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2016.04.19
[debug] Python version 2.7.11 - Darwin-15.5.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 3.0.1, ffprobe 3.0.1, rtmpdump 2.4
[debug] Proxy map: {}
[iqiyi] Get token for logging
[iqiyi] Validate credentials
[iqiyi] temp_id: download video page
[iqiyi] 8e14e8259d848282c48deb8c9a82c90d: Downloading JSON metadata
[iqiyi] 8e14e8259d848282c48deb8c9a82c90d: Downloading video authentication JSON
ERROR: An extractor error has occurred. (caused by KeyError(u'data',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 341, in extract
    return self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/iqiyi.py", line 575, in _real_extract
    data, video_id, _uuid, tvid)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/iqiyi.py", line 458, in construct_video_urls
    't': auth_result['data']['t'],
KeyError: u'data'
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 671, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 347, in extract
    raise ExtractorError('An extractor error has occurred.', cause=e)
ExtractorError: An extractor error has occurred. (caused by KeyError(u'data',)); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Single video: http://www.iqiyi.com/v_19rrlrh3yg.html

---

### Description of your *issue*, suggested solution and other information
My youtube-dl works on other video in Iqiyi, but not this particular page. http://www.iqiyi.com/v_19rrlrh3yg.html. i'm using this account to test youtube-dl  -u 18600775053 -p woaiwojia521 http://www.iqiyi.com/v_19rrlrh3yg.html

**Comment by yan12125** [comment]
Logging with phone numbers are not implemented yet.

**Comment by ckckchen** [comment]
logging with phone number worked for other vip video pages. The above URL is the only page I met this issue

**Comment by yan12125** [comment]
Well in the browser I got the following error:
```
大陸付費內容無法在台灣購買，請切換模式後觀看
```
(Paid contents are not available in Taiwan, please switch the mode)
I enable Unblock-Youku and then it shows:
```
您的電腦可能存在惡意的插件導致視頻無法播放，請檢查並卸載
```
(There may be malicious plugins in your computer that prevent videos from playing. Please check and uninstall them)
Does that mean a China proxy is needed?

**Comment by ckckchen** [comment]
I 'm not sure what happened here. It looks like unblock youku banned by iqiyi.   For most vip contents in iqiyi , they require a Chinese IP address.

**Comment by ckckchen** [comment]
I'll try to find a email formt account to see whether it works or not

**Comment by yan12125** [comment]
Do you have a trustworthy China proxy? Passwords on Iqiyi are sent over unencrypted HTTP channels and I'm afraid passwords would be stolen.

**Comment by yan12125** [comment]
Iqiyi is fixed in 2644e91, while your password is now invalid so I can't test it. Could you try again?

**Comment by yan12125** [comment]
No response from the original poster for a long time. The latest version works fine for me, so closing.
